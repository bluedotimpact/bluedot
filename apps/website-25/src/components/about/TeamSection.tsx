import { Card, SlideList } from '@bluedot/ui';

const teamMembers = [
  {
    imageSrc: '/images/team/dewi.jpg',
    name: 'Dewi Erwan',
    role: 'Co-founder & CEO',
    linkedInUrl: 'https://www.linkedin.com/in/dewierwan/',
  },
  {
    imageSrc: '/images/team/will.jpg',
    name: 'Will Saunter',
    role: 'Co-founder',
    linkedInUrl: 'https://linkedin.com/in/will-saunter',
  },
  {
    imageSrc: '/images/team/adam.jpg',
    name: 'Adam Jones',
    role: 'AI Safety',
    linkedInUrl: 'https://www.linkedin.com/in/domdomegg/',
  },
  {
    imageSrc: '/images/team/josh.jpg',
    name: 'Josh Landes',
    role: 'Community Manager',
    linkedInUrl: 'https://linkedin.com/in/josh-landes12',
  },
  {
    imageSrc: '/images/team/lilian.jpg',
    name: 'Li-Lian Ang',
    role: 'Product',
    linkedInUrl: 'https://linkedin.com/in/anglilian',
  },
  {
    imageSrc: '/images/team/tarin.jpg',
    name: 'Tarin Rickett',
    role: 'Engineering',
    linkedInUrl: 'https://linkedin.com/in/tarinrickett/',
  },
  {
    imageSrc: '/images/team/vio.jpg',
    name: 'Viorica Gheorghita',
    role: 'Product',
    linkedInUrl: 'https://www.linkedin.com/in/vioricagheorghita/',
  },
];

const TeamSection = () => {
  return (
    <SlideList
      title="Our team"
      maxItemsPerSlide={5}
      className="team-section section-body"
    >
      {teamMembers.map((member) => (
        <Card
          key={member.name}
          imageSrc={member.imageSrc}
          title={member.name}
          subtitle={member.role}
          ctaUrl={member.linkedInUrl}
          ctaText="LinkedIn"
          isExternalUrl
          className="team-section__card"
          imageClassName="team-section__card-image h-[300px]"
        />
      ))}
    </SlideList>
  );
};

export default TeamSection;
