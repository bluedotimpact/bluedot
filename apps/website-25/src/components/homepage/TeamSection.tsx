import React from 'react';
import { Card, Section } from '@bluedot/ui';

const TeamSection = () => {
  const teamMembers = [
    {
      imageSrc: '/images/team/adam.jpg',
      name: 'Adam Jones',
      role: 'AI Safety',
      linkedInUrl: 'https://linkedin.com/in/adam',
    },
    {
      imageSrc: '/images/team/dewi.jpg',
      name: 'Dewi Erwan',
      role: 'CEO',
      linkedInUrl: 'https://linkedin.com/in/dewi',
    },
    {
      imageSrc: '/images/team/lilian.jpg',
      name: 'Li-Lian Ang',
      role: 'Product',
      linkedInUrl: 'https://linkedin.com/in/li-lian',
    },
    {
      imageSrc: '/images/team/luke.jpg',
      name: 'Luke Drago',
      role: 'AI Governance',
      linkedInUrl: 'https://linkedin.com/in/luke',
    },
    {
      imageSrc: '/images/team/tarin.jpg',
      name: 'Tarin Rickett',
      role: 'Engineering',
      linkedInUrl: 'https://linkedin.com/in/tarin',
    },
  ];

  return (
    <Section title="Our team">
      <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-8 max-w-[1400px] mx-auto">
        {teamMembers.map((member) => (
          <Card
            key={member.name}
            imageSrc={member.imageSrc}
            title={member.name}
            subtitle={member.role}
            ctaUrl={member.linkedInUrl}
            className="w-full"
          />
        ))}
      </div>
    </Section>
  );
};

export default TeamSection;
