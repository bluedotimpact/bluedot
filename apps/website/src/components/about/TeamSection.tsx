import { Card, Section, SlideList } from '@bluedot/ui';

const teamMembers = [
  {
    imageSrc: '/images/team/dewi.jpg',
    name: 'Dewi Erwan',
    role: 'Co-founder & CEO',
    linkedInUrl: 'https://www.linkedin.com/in/dewierwan/',
  },
  {
    imageSrc: '/images/team/will.jpg',
    name: 'Will Saunter',
    role: 'Co-founder',
    linkedInUrl: 'https://linkedin.com/in/will-saunter',
  },
  {
    imageSrc: '/images/team/lilian.jpg',
    name: 'Li-Lian Ang',
    role: 'Product',
    linkedInUrl: 'https://linkedin.com/in/anglilian',
  },
  {
    imageSrc: '/images/team/adam.jpg',
    name: 'Adam Jones',
    role: 'AI Safety',
    linkedInUrl: 'https://www.linkedin.com/in/domdomegg/',
  },
  {
    imageSrc: '/images/team/josh_v2.jpeg',
    name: 'Josh Landes',
    role: 'Community Manager',
    linkedInUrl: 'https://linkedin.com/in/josh-landes12',
  },
  {
    imageSrc: '/images/team/tarin.jpg',
    name: 'Tarin Rickett',
    role: 'Engineering',
    linkedInUrl: 'https://linkedin.com/in/tarinrickett/',
  },
];
const TeamSection = () => {
  return (
    <Section title="Our team" className="team-section !border-b-0">
      <SlideList
        maxItemsPerSlide={4}
        maxRows={3}
        className="team-section__team"
      >
        {teamMembers.map((member) => (
          <Card
            key={member.name}
            imageSrc={member.imageSrc}
            title={member.name}
            subtitle={member.role}
            ctaUrl={member.linkedInUrl}
            ctaText="LinkedIn"
            className="team-section__card"
            imageClassName="team-section__card-image h-[300px] w-[300px]"
          />
        ))}
      </SlideList>
    </Section>
  );
};

export default TeamSection;
