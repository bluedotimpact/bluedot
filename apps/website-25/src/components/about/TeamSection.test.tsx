import { render } from '@testing-library/react';
import {
  describe,
  expect,
  test,
  vi,
} from 'vitest';
import * as deviceDetect from 'react-device-detect';
import TeamSection from './TeamSection';

describe('TeamSection', () => {
  test('renders desktop as expected', () => {
    const { container } = render(<TeamSection />);
    expect(container).toMatchSnapshot();
    expect(container.querySelector('.team-section--desktop')).not.toBeNull();
  });

  test('renders mobile as expected', () => {
    vi.spyOn(deviceDetect, 'isMobile', 'get').mockReturnValue(true);
    const { container } = render(<TeamSection />);
    expect(container).toMatchSnapshot();
    expect(container.querySelector('.team-section--mobile')).not.toBeNull();
    vi.clearAllMocks();
  });
});
