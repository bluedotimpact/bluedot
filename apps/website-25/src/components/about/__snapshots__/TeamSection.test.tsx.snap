// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TeamSection > renders default as expected 1`] = `
<div>
  <div
    class="section mx-16 my-8 max-w-full overflow-hidden team"
  >
    <div
      class="section__title-container ml-4 flex justify-between items-center gap-4"
    >
      <div
        class="section__content flex-1"
      >
        <h2
          class="section__title text-bluedot-normal text-[48px] mb-4 font-serif font-extrabold leading-none relative after:content-[''] after:absolute after:top-1/2 after:translate-y-[-50%] after:ml-8 after:h-[2px] after:bg-bluedot-normal after:w-[100%]"
        >
          Our team
        </h2>
      </div>
    </div>
    <div
      class="section__body"
    >
      <div
        class="team__grid flex flex-row flex-wrap mt-16 max-w-[1400px] mx-auto"
      >
        <div
          class="card flex flex-col items-start transition-transform duration-200 team__card ml-4 mb-12"
        >
          <div
            class="card__image-container max-w-[323px] max-h-[223px] w-full mb-3"
          >
            <img
              alt="Adam Jones"
              class="card__image w-full max-h-full object-cover rounded-2xl team__card-image"
              src="/images/team/adam.jpg"
            />
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h2
                class="card__title text-2xl font-semibold text-bluedot-darker mb-1"
              >
                Adam Jones
              </h2>
              <p
                class="card__subtitle text-base text-bluedot-black"
              >
                AI Safety
              </p>
            </div>
            <div
              class="card__cta-row flex items-center justify-between mt-auto"
            >
              <a
                class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base px-4 py-2 w-fit cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter card__cta"
                data-testid="cta-link"
                href="https://www.linkedin.com/in/domdomegg/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="cta-button__text"
                >
                  LinkedIn
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="card flex flex-col items-start transition-transform duration-200 team__card ml-4 mb-12"
        >
          <div
            class="card__image-container max-w-[323px] max-h-[223px] w-full mb-3"
          >
            <img
              alt="Dewi Erwan"
              class="card__image w-full max-h-full object-cover rounded-2xl team__card-image"
              src="/images/team/dewi.jpg"
            />
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h2
                class="card__title text-2xl font-semibold text-bluedot-darker mb-1"
              >
                Dewi Erwan
              </h2>
              <p
                class="card__subtitle text-base text-bluedot-black"
              >
                CEO
              </p>
            </div>
            <div
              class="card__cta-row flex items-center justify-between mt-auto"
            >
              <a
                class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base px-4 py-2 w-fit cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter card__cta"
                data-testid="cta-link"
                href="https://www.linkedin.com/in/dewierwan/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="cta-button__text"
                >
                  LinkedIn
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="card flex flex-col items-start transition-transform duration-200 team__card ml-4 mb-12"
        >
          <div
            class="card__image-container max-w-[323px] max-h-[223px] w-full mb-3"
          >
            <img
              alt="Josh Landes"
              class="card__image w-full max-h-full object-cover rounded-2xl team__card-image"
              src="/images/team/josh.jpg"
            />
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h2
                class="card__title text-2xl font-semibold text-bluedot-darker mb-1"
              >
                Josh Landes
              </h2>
              <p
                class="card__subtitle text-base text-bluedot-black"
              >
                Community Manager
              </p>
            </div>
            <div
              class="card__cta-row flex items-center justify-between mt-auto"
            >
              <a
                class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base px-4 py-2 w-fit cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter card__cta"
                data-testid="cta-link"
                href="https://linkedin.com/in/josh-landes12"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="cta-button__text"
                >
                  LinkedIn
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="card flex flex-col items-start transition-transform duration-200 team__card ml-4 mb-12"
        >
          <div
            class="card__image-container max-w-[323px] max-h-[223px] w-full mb-3"
          >
            <img
              alt="Li-Lian Ang"
              class="card__image w-full max-h-full object-cover rounded-2xl team__card-image"
              src="/images/team/lilian.jpg"
            />
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h2
                class="card__title text-2xl font-semibold text-bluedot-darker mb-1"
              >
                Li-Lian Ang
              </h2>
              <p
                class="card__subtitle text-base text-bluedot-black"
              >
                Product
              </p>
            </div>
            <div
              class="card__cta-row flex items-center justify-between mt-auto"
            >
              <a
                class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base px-4 py-2 w-fit cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter card__cta"
                data-testid="cta-link"
                href="https://linkedin.com/in/anglilian"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="cta-button__text"
                >
                  LinkedIn
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="card flex flex-col items-start transition-transform duration-200 team__card ml-4 mb-12"
        >
          <div
            class="card__image-container max-w-[323px] max-h-[223px] w-full mb-3"
          >
            <img
              alt="Tarin Rickett"
              class="card__image w-full max-h-full object-cover rounded-2xl team__card-image"
              src="/images/team/tarin.jpg"
            />
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h2
                class="card__title text-2xl font-semibold text-bluedot-darker mb-1"
              >
                Tarin Rickett
              </h2>
              <p
                class="card__subtitle text-base text-bluedot-black"
              >
                Engineering
              </p>
            </div>
            <div
              class="card__cta-row flex items-center justify-between mt-auto"
            >
              <a
                class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base px-4 py-2 w-fit cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter card__cta"
                data-testid="cta-link"
                href="https://linkedin.com/in/tarinrickett/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="cta-button__text"
                >
                  LinkedIn
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="card flex flex-col items-start transition-transform duration-200 team__card ml-4 mb-12"
        >
          <div
            class="card__image-container max-w-[323px] max-h-[223px] w-full mb-3"
          >
            <img
              alt="Viorica Gheorghita"
              class="card__image w-full max-h-full object-cover rounded-2xl team__card-image"
              src="/images/team/vio.jpg"
            />
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h2
                class="card__title text-2xl font-semibold text-bluedot-darker mb-1"
              >
                Viorica Gheorghita
              </h2>
              <p
                class="card__subtitle text-base text-bluedot-black"
              >
                Product
              </p>
            </div>
            <div
              class="card__cta-row flex items-center justify-between mt-auto"
            >
              <a
                class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base px-4 py-2 w-fit cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter card__cta"
                data-testid="cta-link"
                href="https://www.linkedin.com/in/vioricagheorghita/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="cta-button__text"
                >
                  LinkedIn
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="card flex flex-col items-start transition-transform duration-200 team__card ml-4 mb-12"
        >
          <div
            class="card__image-container max-w-[323px] max-h-[223px] w-full mb-3"
          >
            <img
              alt="Will Saunter"
              class="card__image w-full max-h-full object-cover rounded-2xl team__card-image"
              src="/images/team/will.jpg"
            />
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h2
                class="card__title text-2xl font-semibold text-bluedot-darker mb-1"
              >
                Will Saunter
              </h2>
              <p
                class="card__subtitle text-base text-bluedot-black"
              >
                Co-founder
              </p>
            </div>
            <div
              class="card__cta-row flex items-center justify-between mt-auto"
            >
              <a
                class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base px-4 py-2 w-fit cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter card__cta"
                data-testid="cta-link"
                href="https://linkedin.com/in/will-saunter"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="cta-button__text"
                >
                  LinkedIn
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
